policy/modules/contrib/afterburn.te

```diff
@@ -28,6 +31,11 @@ allow afterburn_t afterburn_runtime_t:dir manage_dir_perms;
 manage_files_pattern(afterburn_t, afterburn_runtime_t, afterburn_runtime_t)
 files_pid_filetrans(afterburn_t, afterburn_runtime_t, dir, "metadata")
 
+manage_dirs_pattern(afterburn_t, afterburn_tmp_t, afterburn_tmp_t)
```

Comment:
Is managing directories required on top of permissions added by files_tmp_filetrans? Will in /run/metadata be only plain files or also other file types - dir, symlink, pipe, socket, ...

Reply:
In /run/metadata, only the directory itself and a single plain file. Is there a more restrictive macro?

Reply:
We also create a directory in /tmp which we mount the config drive on.

Reply:
> In /run/metadata, only the directory itself and a single plain file. Is there a more restrictive macro?

You can remove this manage_dirs_pattern then.

Reply:
I think I see where you're going with this. /run/metadata is handled by afterburn_runtime_t above.

I think I can just add 2 explicit create and rmdir perms for directories of afterburn_tmp_t. I'll give it a go.

Reply:
> We also create a directory in /tmp which we mount the config drive on.

This should already be allowed, not sufficiently? You can use this for a quick (and incomplete) check:
```
f42# macro-expander 'files_tmp_filetrans(afterburn_t, afterburn_tmp_t, dir)'
allow afterburn_t tmp_t:dir { open read getattr lock search ioctl add_name remove_name write };
```

Reply:
From observation it is not allowed, because this is the first thing denied by policy when we try to mount a config drive (it creates the mountpoint in /tmp first). We see this in the audit logs.

Reply:
Think we still need `create` and `rmdir`:

```
type=AVC msg=audit(1740764366.643:270): avc:  denied  { create } for  pid=1988 comm="afterburn" name="afterburn-k4uXUK" scontext=system_u:system_r:afterburn_t:s0 tcontext=system_u:object_r:afterburn_tmp_t:s0 tclass=dir permissive=0
```

I'll add just those.

Reply:
(See top-level comment: also needed getattr, so I switched to manage_dir_perms)